aspnetcore/fundamentals/routing.md

```diff
@@ -1,77 +1,327 @@
-author: ardalis
-description: Discover how ASP.NET Core routing functionality is responsible for mapping an incoming request to a route handler.
+author: rick-anderson
+description: Discover how ASP.NET Core routing is responsible for mapping request URIs to endpoint selectors and dispatching an incoming requests to endpoints.
+monikerRange: '>= aspnetcore-2.1'
 ms.author: riande
 ms.custom: mvc
-ms.date: 10/01/2018
+ms.date: 11/13/2018
 uid: fundamentals/routing
 ---
 # Routing in ASP.NET Core
 
 By [Ryan Nowak](https://github.com/rynowak), [Steve Smith](https://ardalis.com/), and [Rick Anderson](https://twitter.com/RickAndMSFT)
 
-Routing functionality is responsible for mapping an incoming request to a route handler. Routes are defined in the app and configured when the app starts. A route can optionally extract values from the URL contained in the request, and these values can then be used for request processing. Using route information from the app, the routing functionality is also able to generate URLs that map to route handlers. Therefore, routing can find a route handler based on a URL, or find the URL corresponding to a given route handler based on route handler information.
+For the 1.1 version of this topic, download [Routing in ASP.NET Core (version 1.1, PDF)](https://webpifeed.blob.core.windows.net/webpifeed/Partners/Routing_1.x.pdf).
 
-Most apps should choose a basic and descriptive routing scheme so that URLs are readable and meaningful. The default conventional route `{controller=Home}/{action=Index}/{id?}`:
+::: moniker range=">= aspnetcore-2.2"
 
-* Supports a basic and descriptive routing scheme:
-* Is a good starting point for web apps intended for use by browsers.
+Routing is responsible for mapping request URIs to endpoint selectors and dispatching incoming requests to endpoints. Routes are defined in the app and configured when the app starts. A route can optionally extract values from the URL contained in the request, and these values can then be used for request processing. Using route information from the app, routing is also able to generate URLs that map to endpoint selectors.
 
-It's common to add additional terse routes to high traffic areas of the app in specialized situations (for example, blog, ecommerce) using [attribute routing](xref:mvc/controllers/routing#attribute-routing) or dedicated conventional routes.
+::: moniker-end
 
-Web APIs should use attribute routing to model the app's functionality as a set of resources where operations are represented by HTTP verbs. This means that many operations (for example, GET, POST) on the same logical resource will use the same URL. Attribute routing provides a level of control that is needed to carefully design an API's URL-space.
+::: moniker range="= aspnetcore-2.2"
+
+To use the latest routing scenarios in ASP.NET Core 2.2, specify the [compatibility version](xref:mvc/compatibility-version) to the MVC services registration in `Startup.ConfigureServices`:
```

Comment:
I didn't do that for this because the compat version (and the "legacy" option) is only applicable to 2.2. When 3.0 rolls out, these changes will be built-in, and no compat switch will be required to get endpoint routing.

WRT to "legacy" routing (and I'll react to your suggestion below in a sec) ... same thing. My suggestion is that this topic should jettison "legacy" routing language at 3.0. There won't be a valid reason to use `IRouter`-based routing at 3.0 AFAICT ... unless they say otherwise. btw ---- that word choice ---- "legacy" ---- that was Ryan's description of it ... I just used his language from his notes. I guess that if engineering views it as "legacy" tech at 2.2, they'll want to express it that way to the community. It does give one the feel'in that they should move away from it asap. Let me know if you would like it changed in spite of these notions ... I'm really fine either way on it.